# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/l10n/tests/mochitest/dom_localization/test_translateRoots.html
# Commit: 626b40e3aad1
# Full Hash: 626b40e3aad147d59a439b0bd1d5dbd328ad6b1a
# Author: Zibi Braniecki <zbraniecki@mozilla.com>
# Date: 2021-08-02 21:43:56
# Regressor Bug: 1613705
# File Overlap Count: 1
# Description:
#   Bug 1613705 - [localization] part9: Update DOMLocalization to match Localization constructor. r=nika,emilio
#   
#   Depends on D113691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113692
# ==============================================================================

diff -r 1cbab987bb7d -r 626b40e3aad1 dom/l10n/tests/mochitest/dom_localization/test_translateRoots.html
--- a/dom/l10n/tests/mochitest/dom_localization/test_translateRoots.html	Mon Aug 02 09:54:10 2021 +0000
+++ b/dom/l10n/tests/mochitest/dom_localization/test_translateRoots.html	Mon Aug 02 09:54:10 2021 +0000
@@ -7,12 +7,15 @@
   <link rel="stylesheet" type="text/css" href="chrome://mochikit/content/tests/SimpleTest/test.css">
   <script type="application/javascript">
   "use strict";
-  async function* generateBundles(resourceIds) {
-    const bundle = new FluentBundle("en-US");
-    bundle.addResource(new FluentResource("title = Hello World"));
-    bundle.addResource(new FluentResource("title2 = Hello Another World"));
-    yield bundle;
-  }
+  const l10nReg = new L10nRegistry();
+  const fs = [
+    { path: "/localization/en-US/mock.ftl", source: `
+title = Hello World
+title2 = Hello Another World
+` },
+  ];
+  const source = L10nFileSource.createMock("test", ["en-US"], "/localization/{locale}", fs);
+  l10nReg.registerSources([source]);
 
   window.onload = async function() {
     SimpleTest.waitForExplicitFinish();
@@ -20,7 +23,8 @@
     const domLoc = new DOMLocalization(
       [],
       false,
-      { generateBundles },
+      l10nReg,
+      ["en-US"],
     );
 
     const frag1 = document.querySelectorAll("div")[0];
@@ -28,7 +32,7 @@
     const h1 = document.querySelectorAll("h1")[0];
     const h2 = document.querySelectorAll("h2")[0];
 
-    domLoc.addResourceIds(["dummy.ftl"]);
+    domLoc.addResourceIds(["/mock.ftl"]);
     domLoc.connectRoot(frag1);
     domLoc.connectRoot(frag2);
 